### PR TITLE
Fix building against non GPL FFmpeg

### LINKED ISSUE #99: p4a: compile against non GPL ffmpeg version
I try to compile ffpyplayer against non GPL ffmpeg with p4a. I use version n4.3.1 of ffmpeg and v4.3.1 of ffpyplayer.

ffpyplayer build fails during p4a run due to missing ``postprocess.h``, which is not included in built when compiling ffmpeg without GPL components.

>   In file included from ffpyplayer/pic.c:653:	
    ffpyplayer/clib/misc.h:18:10: fatal error: 'libpostproc/postprocess.h' file not	
          found	
    #include "libpostproc/postprocess.h"	
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~

Any ideas?

### PR DESCRIPTION
**Pull request by rnixx**
Fixes #99

### CODE REVIEW

**Comment by rnixx** [comment]
I adopted p4a ffmpeg and ffpyplayer recipes to use recent versions and compile GPL components only if necessary here:

https://github.com/kivy/python-for-android/compare/develop...rnixx:rnixx_custom_versions

the ffmpeg recipe also contains a fix that it builds for android x86.

if you can merge this PR and drop a release i'll point to the official release and prepare a PR for p4a.

Are there any concerns with my changes in this PR?

File: setup.py [hunk 1]
```diff
@@ -206,8 +207,10 @@ def get_paths(name):
     ffmpeg_lib, ffmpeg_include = get_paths('FFMPEG')
     libraries.extend([
         'avcodec', 'avdevice', 'avfilter', 'avformat',
-        'avutil', 'swscale', 'swresample', 'postproc', 'm'
+        'avutil', 'swscale', 'swresample', 'm'
     ])
+    if environ.get('CONFIG_POSTPROC') != '0':
```
Comment (matham) [inline setup.py:212]:
```suggestion
    if config_postproc['config_postproc']:
```

File: setup.py [hunk 2]
```diff
@@ -234,8 +237,12 @@ def get_paths(name):
...
 
     # ffmpeg
-    objects = ['avcodec', 'avdevice', 'avfilter', 'avformat',
-                   'avutil', 'swscale', 'swresample', 'postproc']
+    objects = [
+        'avcodec', 'avdevice', 'avfilter', 'avformat',
+        'avutil', 'swscale', 'swresample'
+    ]
+    if environ.get('CONFIG_POSTPROC') != '0':
+        objects.extend(['postproc'])
```
Comment (matham) [inline setup.py:245]:
```suggestion
    if config_postproc['config_postproc']:
        objects.append('postproc')
```

File: setup.py [hunk 3]
```diff
@@ -206,8 +207,10 @@ def get_paths(name):
     ffmpeg_lib, ffmpeg_include = get_paths('FFMPEG')
     libraries.extend([
         'avcodec', 'avdevice', 'avfilter', 'avformat',
-        'avutil', 'swscale', 'swresample', 'postproc', 'm'
+        'avutil', 'swscale', 'swresample', 'm'
     ])
+    if environ.get('CONFIG_POSTPROC') != '0':
+        libraries.extend(['postproc'])
```
Comment (matham) [inline setup.py:213]:
```suggestion
        libraries.append('postproc')
```

**Review by matham — CHANGES_REQUESTED** [review]

**Comment by matham** [comment]
I plan to make a release soon to add py3.9 support.

File: setup.py (hunk 1 above)
Comment (rnixx) [reply setup.py]: done

File: setup.py (hunk 2 above)
Comment (rnixx) [reply setup.py]: done

File: setup.py (hunk 3 above)
Comment (rnixx) [reply setup.py]: done

**Review by matham — APPROVED** [review]

**Pull request merged**